libkpod/config.go

```diff
@@ -0,0 +1,63 @@
+package libkpod
+
+import (
+	"io/ioutil"
+
+	"github.com/BurntSushi/toml"
+	cstorage "github.com/containers/storage"
+)
+
+// Config stores the configuration options for libkpod
+type Config struct {
+	Root                     string
+	RunRoot                  string
+	Storage                  string
+	StorageOptions           []string
+	LogDir                   string
+	Runtime                  string
+	RuntimeUntrustedWorkload string
+	DefaultWorkloadTrust     string
+	Conmon                   string
+	ConmonEnv                []string
+	CgroupManager            string
+	DefaultTransport         string
+	InsecureRegistries       []string
+	SignaturePolicyPath      string
+}
+
+// FromFile populates the Config from the TOML-encoded file at the given path.
+// Returns errors encountered when reading or parsing the files, or nil
+// otherwise.
+func (c *Config) FromFile(path string) error {
```

Comment:
Maybe it would be simpler to just move most of the CRI-O config in here, then. We will eventually end up reusing every part except APIConfig, I imagine. We can move all the other structs in here, and have a separate version of the config that does not include APIConfig. @14rcole WDYT? If this ends up being harder than the way you're doing it now, I have no objections to this approach.